zulip/zulip
# message_flag_updates: Fix handling of post failures for flag updates.

Previously, when a message's POST of flag update fails, the UI is rendered as if it has succeeded. However, when the site is reloaded, the change could not be seen (for example starring of messages).

This is fixed by storing the POST failures in localstorage,and updating the it when the site reloads and flushing the values which were updated from the localstorage.



[CZO Thread](https://chat.zulip.org/#narrow/stream/137-feedback/topic/Message.20not.20getting.20added.20to.20the.20Starred.20messages.2E)



**Testing Procedure** 
- Terminate the Dev Server.
- Star some message.
- Restart the server.
- The Starred message could still be seen in the Starred list.

<details>
<summary>Self-review checklist</summary>





      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.


Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).


Completed manual review and testing of the following:

</details>
